### DISCUSSION TITLE
Alternative to map 'by expression' for multiple source to target mappings?

### DISCUSSION THREAD

**Discussion by zyberzebra** [mapstruct/mapstruct#3613] [Q&A] [answered]
Am I missing something, or is that the only way to do it:?
```java
@org.mapstruct.Mapper
public interface Mapper {

    @Mapping( target = "funnyCombinationOfIdAndCountry", expression = "java(combine( airplane, cheese ))")
    Cheeseplane map(Airplane airplane, Cheese cheese);

    @Named( "combine" )
    default String combine(Airplane airplane, Cheese cheese){
        return airplane.getAirplaneID() + cheese.getCountry();
    }
}
``` 
![image](https://github.com/mapstruct/mapstruct/assets/73640367/ad47981c-9de4-4772-bf5d-267d00df871b)

**Comment by thunderhook** [comment, marked as answer]
Unfortunately, you're not missing anything. This is not supported at this time. For more information on how to deal with issues like this, see [this Stack Overflow answer](https://stackoverflow.com/a/47705023/1213544) using `@Context`.

So in your case this would look like:

```java
@org.mapstruct.Mapper
public interface Mapper {

    @Mapping( target = "funnyCombinationOfIdAndCountry", source = "airplane", qualifiedByName = "combine" ) // not sure if source = "airplane" is needed
    Cheeseplane map(Airplane airplane, @Context Cheese cheese);

    @Named( "combine" )
    default String combine(Airplane airplane, @Context Cheese cheese){
        return airplane.getAirplaneID() + cheese.getCountry();
    }
}
```

**Reply by zyberzebra** [reply]
Thanks for your answer, guess I have to live with it. 
And yes source is needed, or else is delivers this error: `The type of parameter "airplane" has no property named "funnyCombinationOfIdAndCountry". Please define the source property explicitly.`
